driver-core/src/main/com/mongodb/internal/operation/ChangeStreamBatchCursor.java

```diff
@@ -165,20 +165,20 @@ private void cachePostBatchResumeToken(final AggregateResponseBatchCursor<RawBso
      * @param lastIdConsumer Is {@linkplain Consumer#accept(Object) called} iff {@code rawDocuments} is successfully converted
      *                       and the returned {@link List} is neither {@code null} nor {@linkplain List#isEmpty() empty}.
      */
-    @Nullable
     static <T> List<T> convertAndProduceLastId(@Nullable final List<RawBsonDocument> rawDocuments,
                                                final Decoder<T> decoder,
                                                final Consumer<BsonDocument> lastIdConsumer) {
-        List<T> results = null;
+        List<T> results = new ArrayList<>();
```

Comment:
Note: This follows on from the previous AsyncSingleBatchCursor change where `next()` no longer returns a null value.

Reply:
Never mind, I mis-read your comment.